openpype/pipeline/colorspace.py

```diff
@@ -0,0 +1,116 @@
+import re
+import os
+import platform
+import PyOpenColorIO as ocio
+from openpype.settings import get_project_settings
+from openpype.pipeline import legacy_io
+from openpype.lib.log import Logger
+
+log = Logger.get_logger(__name__)
+
+
+def get_imagio_colorspace_from_filepath(
+    path, host=None, project_name=None, validate=True
+):
+    project_name = project_name or legacy_io.Session["AVALON_PROJECT"]
+    host = host or legacy_io.Session["AVALON_APP"]
+
+    config_path = get_imageio_config(project_name, host)
+    file_rules = get_imageio_file_rules(project_name, host)
+
+    # match file rule from path
+    colorspace_name = None
+    for _frule_name, file_rule in file_rules.items():
+        pattern = file_rule["pattern"]
+        extension = file_rule["ext"]
+        ext_match = re.match(
+            r".*(?=.{})".format(extension), path
+        )
+        file_match = re.search(
+            pattern, path
+        )
+
+        if ext_match and file_match:
+            colorspace_name = file_rule["colorspace"]
+
+    if not colorspace_name:
+        log.info("No imageio file rule matched input path: '{}'".format(
+            path
+        ))
+        return None
+
+    # validate matching colorspace with config
+    if validate and config_path:
+        config_obj = validate_imageio_config_from_path(config_path)
+        validate_imageio_colorspace_in_config(config_obj, colorspace_name)
+
+    return colorspace_name
+
+
+def validate_imageio_colorspace_in_config(config_obj, colorspace_name):
+    if not config_obj.getColorSpace(colorspace_name):
+        raise ocio.Exception(
+            "Missing colorspace '{}' in config file '{}'".format(
+                colorspace_name, config_obj.getWorkingDir())
+        )
+
+
+def validate_imageio_config_from_path(config_path):
```

Comment:
I would expect the validation would return `True/False` instead of reraising (Possibly different error).
```
def create_imageio_config(config_path):
    return ocio.Config.CreateFromFile(config_path)


def validate_imageio_config_from_path(config_path):
    try:
        create_imageio_config(config_path)
    except ocio.Exception:
        return False
    return True
```

Reply:
fixed in https://github.com/pypeclub/OpenPype/pull/4270